### PR TITLE
units: handle multiplication integer overflow

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by niedbalski** [comment]
@brauner thank you for the quick turnaround on this, this partially fixes #7627, but not the original reported issue (yet).

File: shared/units/units.go [hunk 1]
```diff
@@ -5,6 +5,19 @@ import (
...
 )
 
+func handleOverflow(val int64, mult int64) (int64, error) {
+	result := val * mult
+	if val == 0 || mult == 0 || val == 1 || mult == 1 {
+		return result, nil
+	}
+
+	if val != 0 && (result/val) != mult {
+		return 0, fmt.Errorf("Overflow multiplying %d with %d", val, mult)
```
Comment (niedbalski) [inline shared/units/units.go:15]: other error conditions are returning -1, error, why this would return 0, error?

File: shared/units/units.go [hunk 2]
```diff
@@ -5,6 +5,19 @@ import (
 	"strconv"
 )
 
+func handleOverflow(val int64, mult int64) (int64, error) {
+	result := val * mult
+	if val == 0 || mult == 0 || val == 1 || mult == 1 {
```
Comment (niedbalski) [inline shared/units/units.go:10]: is there a case on which case mult can be 0?

File: shared/units/units.go (hunk 1 above)
Comment (brauner) [reply shared/units/units.go:15]: Good catch.

File: shared/units/units.go (hunk 2 above)
Comment (brauner) [reply shared/units/units.go:10]: Not currently but if this is used anywhere else this check can't hurt.

**Pull request merged**